server/sitemap/generators/static.js

```diff
@@ -6,7 +6,7 @@ const { info } = require('../../util/log');
 function getRoutes(haystack, root = '') {
 	return haystack
 		// Remove excluded routes
-		.filter(route => !route.meta?.excludeFromStaticSitemap)
+		.filter(route => !route.meta?.excludeFromStaticSitemap || route.hasOwnProperty('redirect'))
```

Comment:
This should be inverted, because this filter function includes routes when the return value is `true`:
```
route => !route.meta?.excludeFromStaticSitemap && !route.redirect
```